docs/taskcluster-yml-v1.md

```diff
@@ -0,0 +1,189 @@
+---
+title: .taskcluster.yml version 1
+order: 20
+---
+
+Your main interface to Taskcluster-Github is via `.taskcluster.yml` in the root
+of your project. This is a YAML file that speciifies the tasks to run and when.
+
+The format of the file is:
+
+```yaml
+version: 1
+policy:
+  pullRequests: ..
+tasks:
+  - (task definition)
+  - ...
+```
+
+The core of Taskcluster-Github's operation is this: when an event occurs on Github, such as a push or a pull request, it loads `.taskcluster.yml` from the commit specified in the event, renders it with JSON-e, and then calls `Queue.createTask` for each of the specified tasks.
+
+# JSON-e Rendering
+
+The entire YAML file is rendered using
+[JSON-e](https://github.com/taskcluster/json-e). You can view it as a `template`. The following `context` variables are provided:
+
+* `tasks_for` - defines the type of event, one of `github-push`,
+  `github-pull-request` or `github-release`.
+
+* `event` - the raw Github event; see
+  * [PushEvent](https://developer.github.com/v3/activity/events/types/#pushevent)
+  * [PullRequestEvent](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+  * [ReleaseEvent](https://developer.github.com/v3/activity/events/types/#releaseevent)
+
+* `now` - the current time, as a string; this is useful for reproducible `$fromNow` invocations (see JSON-e documentation)
+
+* `as_slugid` - a function which, given a label, will generate a slugid. Multiple calls with the same label for the same event will generate the same slugid, but different slugids in different events.  Use this to generate taskIds, etc.
+
+## Result
+
+After rendering, the resulting data structure should have a `tasks` property containing a list of task definitions. Each task definition should match the [task
+schema](https://docs.taskcluster.net/reference/platform/taskcluster-queue/docs/task-schema) as it will be passed nearly unchanged to `Queue.createTask`, The exception is that the provided task definition must contain a `taskId` field, which the service will remove and pass to `Queue.createTask` directly.
+
+# Policies
+
+The `policy` property defines policies for what is allowed on the repository. Policies are always read from the default branch (generally `master`) of the repository. This prevents a malicious contributor from changing the policy applied to a pull request in the pull request itself.
```

Comment:
Will this involve a JSON-e rendering?  If so, we need to define what the context looks like for that rendering.  If not, then we need to say that the rendering only applies to the `tasks` property, and not the `policy` property.

I'm leaning toward not rendering right now, for simplicity - @imbstack what do you think?

Reply:
Rendering that would be confusing/scary! I agree, let's just render what is in `tasks`.

Reply:
I'll make sure that's clear in the doc. I'll also change the structure a bit and move the Policies section up, so that the rendering and tasks definition became neighbours.